tools/normalize_mri_protocol_range_data.php

```diff
@@ -0,0 +1,113 @@
+    $mp_idx = $DB->pselect("SELECT ID FROM $table_name", array());
+    $idx = array();
+    foreach($mp_idx as $num => $value) {
+        array_push($idx, $value["ID"]);
+    }
+
+// get all column names like "%_range" from mri_protocol table
+    $mp_columns = $DB->pselect("SELECT column_name FROM information_schema.columns WHERE table_name=:table_name", array('table_name' => $table_name));
+    $mp_range_columns = array();
+    foreach($mp_columns as $id => $value) {
+        if (substr($value["column_name"], -5) == "range") {
+            array_push($mp_range_columns, $value["column_name"]);
+        } else if (substr($value["column_name"], -5) == "Range") {
+            array_push($mp_range_columns, $value["column_name"]);
+        }
+    }
+
+    foreach($idx as $id) {
+        $mp_data = $DB->pselect("SELECT * FROM $table_name mp WHERE mp.ID=:id", array('id' => $id));
+        $data_to_insert = array();
+        foreach($mp_range_columns as $range_col) {
+            if (substr($range_col,0, 5) == "Valid") {
+                $col_data = explode("-", $mp_data[0][$range_col]);
+                $col_name = str_replace('Range', '', $range_col);
+                $col_min = $col_name . "min";
+                $col_max = $col_name . "max";
+            } else {
+                $col_data = explode("-", $mp_data[0][$range_col]);
+                $col_name = str_replace('range', '', $range_col);
+                $col_min = $col_name . "min";
+                $col_max = $col_name . "max";
+            }
+            if (sizeof($col_data) > 1) {
+                $data_to_insert[$col_min] = $col_data[0];
+                $data_to_insert[$col_max] = $col_data[1];
+            } else if (sizeof($col_data) == 1 && $col_data[0] != "") {
+                $data_to_insert[$col_min] = $col_data[0];
+                $data_to_insert[$col_max] = $col_data[0];
+            }
+        }
+        $not_null = 0;
+        foreach ($data_to_insert as $data) {
+            if ($data != null) {
+                $not_null += 1;
+            }
+        }
+        // only update non-null range cols
+        if ($not_null > 0) {
+            $DB->update($table_name, $data_to_insert, array('ID' => $id));
+        }
+    }
+
+    echo("$table_name: Data insertion for new min & max columns is complete.\n");
+
+    foreach($mp_range_columns as $range_column) {
+        $output .= "ALTER TABLE $table_name DROP $range_column;\n";
+    }
+
+    if ($printToSQL) {
+        _exportSQL($output);
+    } else {
+        echo("Please execute the following SQL statements to delete the original range columns.\n\n");
+        echo("$output\n");
+    }
+}
+
+$tables_to_normalize = array("mri_protocol", "mri_protocol_checks");
+foreach ($tables_to_normalize as $table) {
+    split_ranges($table, $printToSQL, $output);
+}
+
+function _exportSQL ($output) {
+    //export file
+    $filename = __DIR__ . "/../project/tables_sql/DELETE_mri_protocol_range_columns.sql";
```

Comment:
Is it true that all projects have a directory `__DIR__ . "/../project/tables_sql`? I did not have that directory on my sandbox and got a "No such file or directory for tables_sql" and the DROP statements got lost. Might be worth creating that folder if it does not exist?

Reply:
IBIS has this directory. @ridz1208, is that also the case for CCNA?